python/TestHarness/testers/PetscJacobianTester.py

```diff
@@ -26,6 +26,12 @@ def validParams():
                                           "at every non-linear iteration of every time step. This is only "
                                           "relevant for petsc versions >= 3.9.")
         params.addParam('turn_off_exodus_output', True, "Whether to set exodus=false in Outputs")
+
+        # override default values
+        params.valid['valgrind'] = 'none'
+        params.valid['petsc_version'] = '>=3.9.4'
+        params.valid['method'] = 'opt'
+
```

Comment:
I probably know what you're going to say...but what about removing `difference_tol` and adding in `snes_test_err` while you're tooling around in here?

Reply:
I haven't followed the `snes_test_err` discussion. Why is that approach better?

Reply:
It's not better, it's two different things, sorry for conflating them. `difference_tol` doesn't contain the jacobian check information. Right now `difference_tol` is just set to some number above the `||J - Jfd||_F` by guess and check (so really you could leave it but give it a really high default tolerance.

`snes_test_err` sometimes needs to be adjusted to get `ratio_tol` low. This is now set through cli_args, but feels like it should be a tester option.

 @lindsayad can help clarify.

Reply:
Oh, ok. Is `snes_test_err` the finite differencing parameter?

Reply:
> Oh, ok. Is `snes_test_err` the finite differencing parameter?

Yes. Yea `difference_tol` is highly subject to the scaling of your system.